api.md

```diff
@@ -238,41 +238,55 @@ Add a new data node to the database to be used for creating
+that existing distributed hypertables will not automatically use the
+newly added data node. For existing distributed hypertables to use
+added data nodes, use [`attach_data_node`](#attach_data_node).
+
+If the data node already exists, the command will abort with either an
+error or a notice depending on the value of `if_not_exists`.
+
+If `bootstrap` is true, the function will attempt to bootstrap the
+data node by:
+1. Connecting to the database given in `bootstrap_database`.
+2. Creating the database given in `database` that will serve as the
+   new data node.
+3. Loading the TimescaleDB extension in the new database.
+4. Set the distributed UUID of the database to the UUID of the current
+   database to mark it as a data node belonging to this distributed
+   database.
 
 #### Errors
 
 An error will be given if:
-* an attempt is made to run the function inside a transaction _or_
-* the remote database already exists on the data node and
-  `if_not_exists` is not set.
+* The function is executed inside a transaction.
+* The function is executed in a database that is already a data node.
+* The data node already exists and `if_not_exists` is `FALSE`.
+* The port number is not valid for use as a port number.
+* If `boostrap` is `FALSE` and the remote database did not have the
+  TimescaleDB extension loaded.
+* If `bootstrap` is `FALSE` and the remote database was not owned by
+  the current user.
+* If `bootstrap` is `FALSE` and the remote database did not have the
+  distributed UUID set to the UUID of the current database.
+
 
 #### Required Arguments [](add_data_node-required-arguments)
 
 | Name        | Description                         |
 | ----------- | -----------                         |
 | `node_name` | Name for the data node.             |
+| `host`      | Host name for the remote data node. |
 
 #### Optional Arguments [](add_data_node-optional-arguments)
 
 | Name                 | Description                                           |
 |----------------------|-------------------------------------------------------|
-| `host`               | Host name for the remote data node. The default is `'localhost'`. |
-| `port`               | Port to use on the remote data node. The default is the PostgreSQL port used by the access node on which the function is executed. |
 | `database`           | Database name where remote hypertables will be created. The default is the current database name. |
-| `password`           | Password to be used when connecting to the remote data node. Note that the user name will be the same as the current user. The default is to not use a password. |
-| `if_not_exists`      | Do not fail if the data node already exists. The data node will not be overwritten, but the command will not generate an error and instead generate a notice. |
+| `port`               | Port to use on the remote data node. The default is the PostgreSQL port used by the access node on which the function is executed. |
```

Comment:
Now that we've removed authentication information from the arguments, the docs here are missing a description of how authentication works. We should have a section about how to authenticate with data nodes, either by configuring a password file or setting up certificates.

Reply:
Added this to pivotal: https://www.pivotaltracker.com/story/show/168837146